python/paddle/v2/fluid/framework.py

```diff
@@ -15,6 +15,37 @@ def unique_name(prefix):
     return "_".join([prefix, str(uid)])
 
 
+def convert_np_dtype_to_dtype_(np_dtype):
```

Comment:
The reason this was made a global function is because this is needed outside of the `Variable` class.  In layer_helper, we want to make sure that every parameter which has not been supplied an initializer has a default initializer. This initializer depends on the `dtype` of the parameter. If the parameter is of type float, then `XavierInitializer` is used otherwise the parameter is initialized with Zeros for int and bool types. 

Now we need this method outside because users can also pass np datatypes as dtypes. The initializer needs to be specified in layer_helper and hence we need to check whether the supplied datatype (which could be np.datatype or core.DataType) is of type float. Do you have any suggestion on how to accomplish this without making this global?

Reply:
I find a solution resolve this problem.  `convert_np_dtype_to_dtype_ ` goes the wrong way...  
 this function just makes user can configure a data type string `float32`, `float64`. But we should only let user configure support datatype like `paddle.float32`, `paddle.float64`, and make the real type conversion(from/to numpy) happens in the feed/fetch implementation.